test_runner/src/main/kotlin/ftl/args/ArgsHelper.kt

```diff
@@ -44,6 +45,8 @@ object ArgsHelper {
         YamlObjectMapper().registerKotlinModule()
     }
 
+    private var projectIdSource: String? = null
```

Comment:
`projectIdSource` -> `projectId`

Reply:
and maybe we should move it to `FtlContstans` or make it lazy withing this file

Reply:
Unfortunately not, it's actually the path where project id was read from.

Reply:
> and maybe we should move it to FtlContstans 

It's not used anywhere outside of this class. That would create unnecessary couplings

> or make it lazy val inside this file

`null` is desired here. `projectIdSource` is used when error message is created. Since we do not control where `ServiceOptions.getDefaultProjectId()` found (or from where) project id we should not add `(from ....)` string in that case. I don't know, how `by lazy { }` could be implemented in a reasonable way to provide expected functionality. `lateinit` could work then we would always assign a value (even null) otherwise we would have `UninitializedPropertyAccessException` thrown.